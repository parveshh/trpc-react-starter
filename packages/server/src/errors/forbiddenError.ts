import { BaseError } from "./baseError";

export class ForbiddenError extends BaseError {
  constructor(message: string) {
    super(message, "FORBIDDEN");
  }
}
