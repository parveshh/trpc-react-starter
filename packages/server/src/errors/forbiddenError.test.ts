import { ForbiddenError } from "./forbiddenError";

describe("ForbiddenError", () => {
  it("should have message", () => {
    const message = "Some Error";
    const throwError = () => {
      throw new ForbiddenError(message);
    };

    try {
      throwError();
    } catch (e: ForbiddenError | unknown) {
      if (!(e instanceof ForbiddenError)) {
        throw e;
      }
      expect(e.message).toBe(message);
      expect(e.code).toBe("FORBIDDEN");
      expect(e).toBeInstanceOf(ForbiddenError);
    }
  });
});
