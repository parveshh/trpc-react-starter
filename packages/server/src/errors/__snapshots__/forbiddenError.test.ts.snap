// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForbiddenError should have message 1`] = `"Some Error"`;
